src/math/log_e.c

```diff
@@ -0,0 +1,194 @@
+	num_left_shifts += shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v <<= shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	/*  The input has been left-shifted so the most-significant-bit is a 1. */
+	/*  Reinterpret the output as unsigned with one integer bit, so */
+	/*  that 1 <= x < 2. */
+	x = v << shift_factor;
+	/*  Let Q = int(u).  Then u = Q*2^(-u_fraction_length),	 */
+	/*  and x = Q*2^num_left_shifts * 2^(1-word_length). Therefore, */
+	/*  u = x*2^n, where n is defined as: */
+	/*  Extract the high byte of x */
+	/*  Convert the high byte into an index for lookup table */
+	slice_temp = (v << shift_factor) >> 24;
+	/*  Interpolate between points. */
+	/*  The upper byte was used for the index into lookup table */
+	/*  The remaining bits make up the fraction between points. */
+	a_i = BASE_E_LOG_DENOM_WORDLENGTH - num_left_shifts;
+	a_i <<= 16;
+	l1_i = iv1[(BASE_E_LOG_WRAP_SCHAR_BITS & ((x >> 24) + 129)) - 1];
+	s_v = a_i + l1_i;
+	l2_i = iv1[(BASE_E_LOG_WRAP_SCHAR_BITS & (slice_temp + 130)) - 1];
+	l3_i = iv1[(slice_temp - 127) - 1];
+	a = s_v + (((x & BASE_E_LOG_UPPERBYTES) * (int64_t)(l2_i - l3_i)) >> 24);
+
+	/* calculate denominator-const exp(1)*/
+	shift_factor = iv[denominator >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v = denominator << shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts = shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v <<= shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v <<= shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	/*  The input has been left-shifted so the most-significant-bit is a 1. */
+	/*  Reinterpret the output as unsigned with one integer bit, so */
+	/*  that 1 <= x < 2. */
+	x = v << shift_factor;
+	slice_temp = (v << shift_factor) >> 24;
+	a_i = BASE_E_LOG_NUMER_WORDLENGTH - num_left_shifts;
+	a_i <<= 16;
+	l1_i = iv1[(BASE_E_LOG_WRAP_SCHAR_BITS & ((x >> 24) + 129)) - 1];
+	s_v = a_i + l1_i;
+	l2_i = iv1[(BASE_E_LOG_WRAP_SCHAR_BITS & (slice_temp + 130)) - 1];
+	l3_i = iv1[(slice_temp - 127) - 1];
+	b = s_v + (((x & BASE_E_LOG_UPPERBYTES) * (int64_t)(l2_i - l3_i)) >> 24);
+
+	/*Division 64-bit */
+	logbefxp = (uint32_t)division((int64_t)a << BASE_E_LOG_NUM_SHIFTBITS, (int64_t)b);
```

Comment:
If I change this line to simpler version the x86 and xtensa cmocka tests are still passed.

`logbefxp = (uint32_t)(((int64_t)a << BASE_E_LOG_NUM_SHIFTBITS) / b);
`